pin/views.py

```diff
@@ -1,4 +1,316 @@
-from django.shortcuts import render
-from django.views       import View
-from django.http        import JsonResponse
+import boto3
+import json
+import jwt
 
+from django.shortcuts       import render
+from django.views           import View
+from django.http            import JsonResponse, HttpResponse
+from django.forms.models    import model_to_dict
+from django.core            import serializers
+from django.db.models       import Q
+from django.core.exceptions import ObjectDoesNotExist
+
+from pin.models         import (
+    Topic,
+    UserTopic,
+    Board,
+    Pin,
+    Topic,
+    PinUser,
+    PinTopic,
+    Comment
+)
+from account.models         import User, Follow, Social
+from pinterrorist.settings  import S3URL, SECRET_KEY, ACCESS_KEY, SECRET_ACCESS_KEY
+
+def login_required(func):
+        def wrapper(self, request, *args, **kwargs):
+            
+            header_token            = request.headers['Authorization']
+            print(header_token)
+            if header_token         == None:
+                return JsonResponse({"message": "THE TOKEN IS EMPTY"}, status=402)
+            decoded_token           = jwt.decode(header_token, SECRET_KEY, algorithm='HS256')['id']
```

Comment:
변수명이 decoded_token 이므로 여기에 id를 저장하는것보다 41에서 decoded_token['id'] 로 하는게 코드를 읽을 때 좋은것같습니다.
41에서도 id필드에 decoded_token이 들어가서 위로 올려서 찾아봤습니다.
```suggestion
            decoded_token           = jwt.decode(header_token, SECRET_KEY, algorithm='HS256')
```